mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -777,15 +778,34 @@ class ALSSuite
   }
 }
 
-class ALSCleanerSuite extends SparkFunSuite {
+class ALSCleanerSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  private var localDir: File = _
```

Comment:
@srowen 
Once'spark.local.dir' is set. it is cached, Utils.deleteRecursively(localDir) is not delete cache.
After running the ALSSuite unit test.  will recusively a TMP spark_*  directory is not delete.
thanks.